TestStartEtcdWrongToken fails if another etcd is running

On Debian CI (see https://ci.debian.net/data/autopkgtest/unstable/amd64/e/etcd/5695459/log.gz) using etcd-3.2.26 for now (but there is no indication this should be different for master), `TestStartEtcdWrongToken` fails if another etcd is running already:

```
=== RUN   TestStartEtcdWrongToken
    TestStartEtcdWrongToken: serve_test.go:36: expected auth: invalid auth options, got listen tcp 127.0.0.1:2380: bind: address already in use
--- FAIL: TestStartEtcdWrongToken (0.00s)
FAIL
FAIL	github.com/coreos/etcd/embed	0.014s
```

I think all(?) the other tests use random/incremental ports and not the default port. Should this be the case here as well, or is this not possible on principle?

@jingyih I have opened PR but looks like some of the TravisCI runs failed and insight into the failures 
* [4142.3](https://travis-ci.com/github/etcd-io/etcd/jobs/383428505) -> --- FAIL: TestBalancerUnderNetworkPartitionWatchLeader (3.25s), this is an unrelated test to my changes. 
* [4142.2](https://travis-ci.com/github/etcd-io/etcd/jobs/383428503) -> Not sure why this failed from reading the logs, need some guidance on what failed here. 

Please assist so I can know what failed and update PR accordingly, thanks!

@teddylear Thanks for your PR!
> 4142.3 -> --- FAIL: TestBalancerUnderNetworkPartitionWatchLeader (3.25s), this is an unrelated test to my changes.

You are right, this test failure is not related to your PR.


> 4142.2 -> Not sure why this failed from reading the logs, need some guidance on what failed here.

Here it says gofmt check failed. https://travis-ci.com/github/etcd-io/etcd/jobs/383428503#L362
Please use gofmt tool to format the go files you changed. (The code indentation looks incorrect to me)
```
gofmt -s -w ./the/path/to/file
```

--- a/.gitignore
+++ b/.gitignore
@@ -7,6 +7,7 @@
 /bin
 *.etcd
 *.log
+*.swp
 /etcd
 /hack/insta-discovery/.env
 *.coverprofile

--- a/embed/serve_test.go
+++ b/embed/serve_test.go
@@ -15,7 +15,9 @@
 package embed
 
 import (
+	"fmt"
 	"io/ioutil"
+	"net/url"
 	"os"
 	"testing"
 
@@ -25,14 +27,38 @@ import (
 // TestStartEtcdWrongToken ensures that StartEtcd with wrong configs returns with error.
 func TestStartEtcdWrongToken(t *testing.T) {
 	tdir, err := ioutil.TempDir(os.TempDir(), "token-test")
+
 	if err != nil {
 		t.Fatal(err)
 	}
 	defer os.RemoveAll(tdir)
+
 	cfg := NewConfig()
+
+	// Similar to function in integration/embed/embed_test.go for setting up Config.
+	urls := newEmbedURLs(2)
+	curls := []url.URL{urls[0]}
+	purls := []url.URL{urls[1]}
+	cfg.LCUrls, cfg.ACUrls = curls, curls
+	cfg.LPUrls, cfg.APUrls = purls, purls
+	cfg.InitialCluster = ""
+	for i := range purls {
+		cfg.InitialCluster += ",default=" + purls[i].String()
+	}
+	cfg.InitialCluster = cfg.InitialCluster[1:]
 	cfg.Dir = tdir
 	cfg.AuthToken = "wrong-token"
+
 	if _, err = StartEtcd(cfg); err != auth.ErrInvalidAuthOpts {
 		t.Fatalf("expected %v, got %v", auth.ErrInvalidAuthOpts, err)
 	}
+}
+
+func newEmbedURLs(n int) (urls []url.URL) {
+	scheme := "unix"
+	for i := 0; i < n; i++ {
+		u, _ := url.Parse(fmt.Sprintf("%s://localhost:%d%06d", scheme, os.Getpid(), i))
+		urls = append(urls, *u)
+	}
+	return urls
 }

